sphinx/ext/autosummary/generate.py

```diff
@@ -126,12 +127,19 @@ def render(self, template_name, context):
 
 # -- Generating output ---------------------------------------------------------
 
-def generate_autosummary_docs(sources, output_dir=None, suffix='.rst',
-                              warn=_simple_warn, info=_simple_info,
-                              base_path=None, builder=None, template_dir=None,
-                              imported_members=False, app=None):
-    # type: (List[str], str, str, Callable, Callable, str, Builder, str, bool, Any) -> None
-
+def generate_autosummary_docs(sources,                 # type: List[str]
+                              output_dir=None,         # type: str
+                              suffix='.rst',           # type: str
+                              warn=_simple_warn,       # type: Callable
+                              info=_simple_info,       # type: Callable
+                              base_path=None,          # type: str
+                              builder=None,            # type: Builder
+                              template_dir=None,       # type: str
+                              imported_members=False,  # type: bool
+                              recursive=False,         # type: bool
```

Comment:
Reordering arguments is always breaking change. Could you add a new option to the tail please? I'm afraid that 3rd party application might give parameter as positional.

Reply:
Done